src/librustdoc/doctest.rs

```diff
@@ -1123,17 +1123,17 @@ impl<'a, 'hir, 'tcx> HirCollector<'a, 'hir, 'tcx> {
         }
 
         attrs.unindent_doc_comments();
-        // The collapse-docs pass won't combine sugared/raw doc attributes, or included files with
-        // anything else, this will combine them for us.
-        if let Some(doc) = attrs.collapsed_doc_value() {
+        // `doc_value()` doesn't combine sugared/raw doc attributes, this will combine them for us.
+        let docs = attrs.collapsed_doc_value();
+        if !docs.is_empty() {
```

Comment:
Are we sure this slight behavior change here and elsewhere doesn't matter? Before, this checked for if there were no DocFragments; now, it checks if the collapsed docs is an empty string.

I think it's probably "more correct" now, but I just wanted to double-check.

Reply:
If you look at https://github.com/rust-lang/rust/blob/c7476adc38bc9348398afcf24117d7d48e7ba9f1/src/librustdoc/html/markdown.rs#L732, it doesn't do anything for empty strings any way; having a check here at all is a microoptimization we could probably remove. So the change shouldn't have an impact.

Reply:
But this code is looking for doctests, not rendering Markdown.

Reply:
I don't _think_ there should be a behavior change, but there is a very slight possibility.

Reply:
This `if` calls `find_testable_code` immediately below. The parser there is parsing the code for doctests.